test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -162,6 +161,24 @@ object TestAspect extends TimeoutVariants {
         effect *> test
     }
 
+  /**
+   * An aspect that runs each test on a separate fiber and prints a fiber dump
+   * if the test fails or has not termianted within the specified duration.
+   */
+  def diagnose(duration: Duration): TestAspectAtLeastR[Live] =
+    new PerTest.AtLeastR[Live] {
+      def perTest[R <: Live, E](test: ZIO[R, TestFailure[E], TestSuccess]): ZIO[R, TestFailure[E], TestSuccess] = {
+        def dump[E, A](fiber: Fiber.Runtime[E, A]): ZIO[Live, Nothing, Unit] =
+          fiber.dump.flatMap(_.prettyPrintM).flatMap(s => Live.live(console.putStrLn(s)))
```

Comment:
Could be nice to prefix this with the test’s label, if possible.

Reply:
Good idea! Will add.